pkg/solana/txm/pendingtx.go

```diff
@@ -47,6 +48,18 @@ type PendingTxContext interface {
 	GetTxState(id string) (TxState, error)
 	// TrimFinalizedErroredTxs removes transactions that have reached their retention time
 	TrimFinalizedErroredTxs() int
+	// GetSignatureInfo returns the transaction ID and TxState for the provided signature
+	GetSignatureInfo(sig solana.Signature) (txInfo, error)
+	// UpdateSignatureStatus updates the status of a signature in the SigToTxInfo map
+	UpdateSignatureStatus(sig solana.Signature, status TxState) error
+	// TxHasReorg determines whether a reorg has occurred for a given tx.
+	// It achieves this by comparing the highest aggregated state across all associated signatures with the current state of the transaction.
+	// If the highest aggregated state is less than the current state, a reorg has occurred and we need to handle it.
+	TxHasReorg(id string) bool
```

Comment:
nit: rename to IsTxReorged

The previous name, to me, looked like was telling this interface that a reorg has occurred.